javascript/packages/orchestrator/src/networkNode.ts

```diff
@@ -538,12 +539,17 @@ export class NetworkNode implements NetworkNodeInterface {
     return spanNames;
   }
 
-  // prevent to seach in the same log line twice.
-  _dedupLogs(logs: string[], useIndex = false): string[] {
-    if (!this.lastLogLineCheckedTimestamp) return logs;
-    if (useIndex) return logs.slice(this.lastLogLineCheckedIndex);
-
-    const lastLineTs = this.lastLogLineCheckedTimestamp.split(" ")[0];
+  // prevent to search in the same log line twice.
+  _dedupLogs(
+    logs: string[],
+    useIndex = false,
+    lastLogLineCheckedTimestamp: string,
+    lastLogLineCheckedIndex: number,
+  ): string[] {
+    if (!lastLogLineCheckedTimestamp) return logs;
+    if (useIndex) return logs.slice(lastLogLineCheckedIndex);
```

Comment:
Here, lastLogLineCheckedIndex is undefined when entering the above loop

Reply:
Well, those two variables work in tandem to represent the different providers features. The invariant here is that `lastLogLineCheckedIndex` will be undefined at the same time that `lastLogLineCheckedTimestamp`  is undefinded, so the this if (https://github.com/paritytech/zombienet/pull/977/files#diff-fc4624d6bb61da0712b154b5018f73681adaf8edadb4236a0b5293ea915ef41eR549) cover this case.

I think there is room to improving here, but we can do it in a follow up pr.

Thanks!